util/osuser/user.go

```diff
@@ -51,12 +51,6 @@ func LookupByUsername(username string) (*user.User, error) {
 type lookupStd func(string) (*user.User, error)
 
 func lookup(usernameOrUID string, std lookupStd, wantShell bool) (*user.User, string, error) {
-	// TODO(awly): we should use genet on more platforms, like FreeBSD.
-	if runtime.GOOS != "linux" {
```

Comment:
Per https://go.dev/doc/install/source#environment
> Choices for $GOOS are android, darwin, dragonfly, freebsd, illumos, ios, js, linux, netbsd, openbsd, plan9, solaris, wasip1, and windows.

We should at least filter out `windows`, `js`, and `wasip1`, which do not have `getent` AFAIK

Reply:
On all those platforms, the existing code that checks for `getent` command failure will catch the "executable file not found" error and fall back to the appropriate standard library code: https://github.com/tailscale/tailscale/blob/8d7b78f3f795e781d939750893610639b224d81a/util/osuser/user.go#L121-L126

IMO, it would be better to not have additional code checking explicit `GOOS` values when there is already a decent failure path, but I'm open to implementing either way. What do you think?

Reply:
It's true that it would work on those platforms, but it creates unnecessary log spam (which is already a noticeable problem with tailscaled), plus the penalty of a `PATH` lookup of a binary we know won't exist.

Let's still check `GOOS` to avoid those.